### PR TITLE
Graph view for hours reports 94354764

### PR DESCRIPTION
**Pull request by brookekfox**

### CODE REVIEW

**Comment by IdahoEv** [comment]
&nbsp;<b>Code Climate</b> has <a href="https://codeclimate.com/repos/52793e167e00a4591d00005c/pull/143">analyzed this pull request</a>.

File: spec/queries/work_unit_query_spec.rb.orig [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    @user_2 = FactoryGirl.create(:user, :name => 'Foo Bar 2')
+    @project_1 = FactoryGirl.create(:project)
+    @project_2 = FactoryGirl.create(:project)
+
+    @recent_work_unit_1 = FactoryGirl.create(:work_unit,
+                           :hours => 5,
+                           :user => @user_1,
+                           :project => @project_1,
+                           :start_time => Time.now - 2.weeks,
+<<<<<<< HEAD
```
Comment (patriciaho) [inline spec/queries/work_unit_query_spec.rb.orig]: These merge conflicts need to be resolved.

File: app/views/hours_reports/index.html.haml.orig [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+= form_tag hours_reports_path do
+  %div.input-container
+    %p Start Date
+    %input.american_date_entry#start-datepicker{:type => "text", :name => "start_date", :value => @start_date}
+  %div.input-container
+    %p End Date
+    %input.american_date_entry#end-datepicker{:type => "text", :name => "end_date", :value => @end_date}
+  = submit_tag "Submit", :id => 'datepicker-submit-btn'
+
+<<<<<<< HEAD
```
Comment (patriciaho) [inline app/views/hours_reports/index.html.haml.orig]: merge conflict

File: app/views/hours_reports/index.html.haml.orig [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+              %div.total= WorkUnitQuery.new(u,@end_date,w,'total').hours
+  = tp.add_tab('Graph') do
+    #placeholder{ :style => 'width:750px;height:450px;' }
+
+:javascript
+  $(function() {
+    var hoursDataset = #{hours_reports_data(@users)};
+    $.plot("#placeholder", hoursDataset);
+  });
+=======
```
Comment (patriciaho) [inline app/views/hours_reports/index.html.haml.orig]: here

File: app/views/hours_reports/index.html.haml.orig [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    %tr.user_row
+      %td
+        %h2= u.name
+      -#- (1..number_of_weeks_between(@start_date, @end_date)+1).each do |w|
+      - @sundays.each do |sunday|
+        %td
+          %div.billable= WorkUnitQuery.new(u,sunday,'billable').hours
+          %div.unbillable= WorkUnitQuery.new(u,sunday,'unbillable').hours
+          %div.total= WorkUnitQuery.new(u,sunday,'total').hours
+>>>>>>> hours_reports_date_selector_93493736
```
Comment (patriciaho) [inline app/views/hours_reports/index.html.haml.orig]: and here

File: spec/queries/work_unit_query_spec.rb.orig [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+		@recent_work_unit_2 = FactoryGirl.create(:work_unit,
+=======
+                           :stop_time => Time.now - 2.weeks + 5.hour),
+        FactoryGirl.create(:work_unit,
+>>>>>>> fdb0c3ef2e0bbcb499f9dc80a268716d8c50282d
+                           :hours => 5,
+                           :user => @user_1,
+                           :project => @project_1,
+                           :start_time => Time.now - 3.weeks,
+<<<<<<< HEAD
```
Comment (patriciaho) [inline spec/queries/work_unit_query_spec.rb.orig]: here

File: spec/queries/work_unit_query_spec.rb.orig [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    @billable_work_units_2 = WorkUnitQuery.new(@user_2,Time.now - 2.weeks,'billable')
+    @unbillable_work_units_1 = WorkUnitQuery.new(@user_1,Time.now - 2.weeks,'unbillable')
+    @unbillable_work_units_2 = WorkUnitQuery.new(@user_1,Time.now - 3.weeks,'unbillable')
+    @total_work_units_1 = WorkUnitQuery.new(@user_1,Time.now - 3.weeks,'total')
+	end
+
+	it "should find older billable hours" do
+		expect(WorkUnitQuery.new(@user_2,Time.now - 6.weeks,'billable').hours).to eq(6.0)
+	end
+=======
```
Comment (patriciaho) [inline spec/queries/work_unit_query_spec.rb.orig]: here

File: spec/queries/work_unit_query_spec.rb.orig [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+                           :stop_time => Time.now - 7.weeks + 5.hour,
+                           :billable => false)
+    ]
+    @billable_work_unit_1 = WorkUnitQuery.new(@user_1,Time.now,2,'billable')
+    @billable_work_unit_2 = WorkUnitQuery.new(@user_2,Time.now,2,'billable')
+    @unbillable_work_unit_1 = WorkUnitQuery.new(@user_1,Time.now,2,'unbillable')
+    @unbillable_work_unit_2 = WorkUnitQuery.new(@user_1,Time.now,3,'unbillable')
+    @total_work_unit_1 = WorkUnitQuery.new(@user_1,Time.now,3,'total')
+  end
+>>>>>>> fdb0c3ef2e0bbcb499f9dc80a268716d8c50282d
```
Comment (patriciaho) [inline spec/queries/work_unit_query_spec.rb.orig]: and here

**Comment by patriciaho** [comment]
I have 4 failing tests here:

```
Failures:

  1) WorkUnitQuery should find recent hours
     Failure/Error: @billable_work_units_1   = WorkUnitQuery.new(@user_1,'Apr 06 15','Apr 19 15','billable').hours
     NoMethodError:
       undefined method `cwday' for "Apr 06 15":String
     # ./app/queries/work_unit_query.rb:7:in `initialize'
     # ./spec/queries/work_unit_query_spec.rb:54:in `new'
     # ./spec/queries/work_unit_query_spec.rb:54:in `block (2 levels) in <top (required)>'

  2) WorkUnitQuery should find older hours
     Failure/Error: @billable_work_units_1   = WorkUnitQuery.new(@user_1,'Apr 06 15','Apr 19 15','billable').hours
     NoMethodError:
       undefined method `cwday' for "Apr 06 15":String
     # ./app/queries/work_unit_query.rb:7:in `initialize'
     # ./spec/queries/work_unit_query_spec.rb:54:in `new'
     # ./spec/queries/work_unit_query_spec.rb:54:in `block (2 levels) in <top (required)>'

  3) Admin views the hours reports should show the last six Sundays as column headers
     Failure/Error: expect(page).to have_content('Mar 22 15')
       expected to find text "Mar 22 15" in "TypeError in HoursReports#index Showing /Users/Patricia/LRD/TimePulse/app/views/hours_reports/index.html.haml where line #32 raised: no implicit conversion of DateTime into String Extracted source (around line #32): 29 30 31 32 33 34 35 %td - @sundays.each do |sun| %td %h2= 'Week ending ' + sun - @users.each do |u| %tr.user_row %td Rails.root: /Users/Patricia/LRD/TimePulse Application Trace | Framework Trace | Full Trace app/views/hours_reports/index.html.haml:32:in `+' app/views/hours_reports/index.html.haml:32:in `block (3 levels) in _app_views_hours_reports_index_html_haml__2416141936585585441_70146012627960' app/views/hours_reports/index.html.haml:30:in `each' app/views/hours_reports/index.html.haml:30:in `block (2 levels) in _app_views_hours_reports_index_html_haml__2416141936585585441_70146012627960' app/views/hours_reports/index.html.haml:26:in `block in _app_views_hours_reports_index_html_haml__2416141936585585441_70146012627960' app/views/hours_reports/index.html.haml:25:in `_app_views_hours_reports_index_html_haml__2416141936585585441_70146012627960' app/controllers/hours_reports_controller.rb:9:in `index' Request Parameters: None Toggle session dump Toggle env dump Response Headers: None"
     # ./spec/stories/admin_views_hours_reports_spec.rb:75:in `block (2 levels) in <top (required)>'

  4) db:sample_data namespace rake task db:sample_data:load should create invoices
     Failure/Error: client.invoices.first.work_units.size.should > 5
       expected: > 5
            got:   5
     # ./spec/tasks/sample_data_rake_spec.rb:80:in `block (4 levels) in <top (required)>'
     # ./spec/tasks/sample_data_rake_spec.rb:78:in `block (3 levels) in <top (required)>'
```

**Pull request merged**